pkgs/applications/editors/poke/default.nix

```diff
@@ -85,10 +104,11 @@ in stdenv.mkDerivation rec {
   meta = with lib; {
     description = "Interactive, extensible editor for binary data";
     homepage = "http://www.jemarch.net/poke";
+    changelog = "https://git.savannah.gnu.org/cgit/poke.git/plain/ChangeLog?h=releases/poke-${version}";
     license = licenses.gpl3Plus;
     maintainers = with maintainers; [ AndersonTorres kira-bruneau ];
     platforms = platforms.unix;
-    changelog = "https://git.savannah.gnu.org/cgit/poke.git/plain/ChangeLog?h=releases/poke-${version}";
+    badPlatforms = [ "aarch64-darwin" ]; # Undefined symbols for architecture arm64
```

Comment:
TIL badPlatforms. But if it's expected to work and fails building, should it be marked `broken` instead here?

Reply:
As far as I know `badPlatforms` is just a nicer & more declarative way to mark a package as broken on a specific platform. The only difference is if you want to force build it you need to specify `NIXPKGS_ALLOW_UNSUPPORTED_SYSTEM` instead of `NIXPKGS_ALLOW_BROKEN`.

Reply:
I mean the difference is semantic - this package is supported, but is broken on that system. (i.e. someone can attempt fix it without porting the app)

Reply:
From what I can tell, it doesn't look this is a Nix packaging problem and I don't think it can be fixed without patching poke. I also don't think there's really much of a semantic difference between a supported platform and a broken platform when the project doesn't mention support for `aarch64-darwin`.

Reply:
`broken` is a Boolean value. I treat `broken = isDarwin` as _"well, it is not working because of some Darwin quirk"_.

On the other hand `badPlatforms` is because the software is not made to build on it. An example would be run ZSNES on an ARM machine - because ZSNES is made in x86 assembly.

As far as I remember, badPlatforms is used by Hydra in order to not try tro build on those bad platforms.

Reply:
Ok, I've been convinced :sweat_smile:, I opened a PR for this here: #183873.